src/main/java/Duke/task/task.java

```diff
@@ -0,0 +1,36 @@
+package Duke.task;
+
+import java.io.Serializable;
+
+public class task implements Serializable {
+    private String message;
+    private boolean state = false;
+    public task(String msg){
+        this.message = msg;
+        if (msg.length()==0){
+            throw new IllegalArgumentException("input can't be empty");
+        }
+    }
+    public void mark(){
+        state = true;
+    }
+
+    public void unmark(){
+        state = false;
+    }
+    protected boolean access_state(){
+        return state;
+    }
+    protected String access_message(){
+        return message;
+    }
+    @Override
+    public String toString(){
```

Comment:
Try to add a white space between method name and curly brazes to improve readability.